tensorboard/plugins/interactive_inference/interactive_inference_plugin.py

```diff
@@ -0,0 +1,405 @@
+
+      serving_bundle = inference_utils.ServingBundle(
+          request.args.get('inference_address'),
+          request.args.get('model_name'), request.args.get('model_type'),
+          request.args.get('model_version'),
+          request.args.get('model_signature'))
+      indices_to_infer = sorted(self.updated_example_indices)
+      examples_to_infer = [self.examples[index] for index in indices_to_infer]
+
+      # Get inference results proto and combine with indices of inferred
+      # examples and respond with this data as json.
+      inference_result_proto = oss_utils.call_servo(
+          examples_to_infer, serving_bundle)
+      new_inferences = inference_utils.wrap_inference_results(
+          inference_result_proto)
+      infer_json = json_format.MessageToJson(
+          new_inferences, including_default_value_fields=True)
+      infer_obj = json.loads(infer_json)
+      resp = {'indices': indices_to_infer, 'results': infer_obj}
+      self.updated_example_indices = set()
+      return http_util.Respond(request, {'inferences': json.dumps(resp),
+                                         'vocab': json.dumps(label_vocab)},
+                               'application/json')
+    except common_utils.InvalidUserInputError as e:
+      return http_util.Respond(request, {'error': e.message},
+                               'application/json')
+    except AbortionError as e:
+      return http_util.Respond(request, {'error': e.details},
+                               'application/json')
+
+  def create_sprite_image(self, examples):
+    """Returns an encoded sprite image for use in Facets Dive.
+
+    Args:
+      examples: A list of serialized example protos to get images for.
+
+    Returns:
+      An encoded PNG.
+    """
+
+    def generate_image_from_thubnails(thumbnails, thumbnail_dims):
+      """Generates a sprite atlas image from a set of thumbnails."""
+      num_thumbnails = tf.shape(thumbnails)[0].eval()
+      images_per_row = int(math.ceil(math.sqrt(num_thumbnails)))
+      thumb_height = thumbnail_dims[0]
+      thumb_width = thumbnail_dims[1]
+      master_height = images_per_row * thumb_height
+      master_width = images_per_row * thumb_width
+      num_channels = 3
+      master = np.zeros([master_height, master_width, num_channels])
+      for idx, image in enumerate(thumbnails.eval()):
+        left_idx = idx % images_per_row
+        top_idx = int(math.floor(idx / images_per_row))
+        left_start = left_idx * thumb_width
+        left_end = left_start + thumb_width
+        top_start = top_idx * thumb_height
+        top_end = top_start + thumb_height
+        master[top_start:top_end, left_start:left_end, :] = image
+      return tf.image.encode_png(master)
+
+    with tf.Session():
+      keys_to_features = {
+          self.image_feature_name:
+              tf.FixedLenFeature((), tf.string, default_value=''),
+      }
+      parsed = tf.parse_example(examples, keys_to_features)
```

Comment:
@nfelt is there any other way to do this without directly involving tf?

Reply:
why the hesitation to use tf? there are other python libraries that can do this image stitching but i don't want to add new deps to TB backend when not necessary.